### PR TITLE
Refactor TestHelper isCSharp to AnalyzerLanguage

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
`Squash ExplodedNode` commit doesn't need a review

Follow up for https://github.com/SonarSource/sonar-dotnet/pull/5108#discussion_r760438989

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs [hunk 1]
```diff
@@ -49,37 +49,43 @@ internal static class TestHelper
...
-        public static (SyntaxTree, SemanticModel) Compile(string snippet, bool isCSharp = true, params MetadataReference[] additionalReferences) =>
-            Compile(snippet, false, isCSharp, additionalReferences);
+        public static (SyntaxTree Tree, SemanticModel Model) CompileCS(string snippet, params MetadataReference[] additionalReferences) =>
+            Compile(snippet, false, AnalyzerLanguage.CSharp, additionalReferences);
 
-        public static (SyntaxTree, SemanticModel) Compile(string snippet, bool ignoreErrors, bool isCSharp, params MetadataReference[] additionalReferences)
+        public static (SyntaxTree Tree, SemanticModel Model) CompileVB(string snippet, params MetadataReference[] additionalReferences) =>
+            Compile(snippet, false, AnalyzerLanguage.VisualBasic, additionalReferences);
+
+        public static (SyntaxTree Tree, SemanticModel Model) Compile(string snippet, bool ignoreErrors, AnalyzerLanguage language, params MetadataReference[] additionalReferences)
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs:61]: `Compile` and `CompileCfg` methods should be private as there are other ` AnalyzerLanguages` that are not handled in the method. Where those methods are used now the CS and VB variants could be used to replace it.

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
One single comment left.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs:61]: I've replaced the single usage that could be simplified.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
Please try to improve the coverage, otherwise good to go!

**Pull request merged**